src/spaces/utilities.jl

```diff
@@ -233,6 +233,75 @@ walk!(agent, ::typeof(rand), model::ABM{<:AbstractGridSpace{D}}; kwargs...) wher
 walk!(agent, ::typeof(rand), model::ABM{<:ContinuousSpace{D}}) where {D} =
     walk!(agent, Tuple(2.0 * rand(model.rng) - 1.0 for _ in 1:D), model)
 
+
+"""
+    rotate(w::SVector{2}, θ)
+Rotate two-dimensional vector `w` by an angle `θ`.
+"""
+rotate(w::SVector{2}, θ) = Angle2d(θ) * w
+
+"""
+    rotate(w::SVector{3}, θ, ϕ)
+Rotate three-dimensional vector `w` by angles `θ` (polar) and `ϕ` (azimuthal).
+"""
+function rotate(w::SVector{3}, θ, ϕ)
+    # find a vector normal to w
+    m = findfirst(w .≠ 0)
+    n = m%3 + 1
+    u = zeros(3)
+    u[n], u[m] = w[m], -w[n]
+    # rotate w around u by the polar angle θ
+    a = AngleAxis(θ, u...) * w
+    # rotate a around the original vector w by the azimuthal angle ϕ
+    AngleAxis(ϕ, w...) * a
+end # function
+
+"""
+    randomwalk!(agent, model, r)
+Move `agent` for a distance `r` in a random direction, respecting periodic 
+boundary conditions and space metric.
+"""
```

Comment:
This documentation string does not mention the extra keywords you mention in the pull request